internal/trigger/http/rest.go

```diff
@@ -98,27 +109,33 @@ func (trigger *Trigger) requestHandler(writer http.ResponseWriter, r *http.Reque
 		Payload:       data,
 	}
 
-	messageError := trigger.Runtime.ProcessMessage(appContext, envelope, trigger.Runtime.GetDefaultPipeline())
-	if messageError != nil {
-		// ProcessMessage logs the error, so no need to log it here.
-		writer.WriteHeader(messageError.ErrorCode)
-		_, _ = writer.Write([]byte(messageError.Err.Error()))
-		return
-	}
+	appContext := trigger.serviceBinding.BuildContext(envelope)
 
-	if len(appContext.ResponseContentType()) > 0 {
-		writer.Header().Set(common.ContentType, appContext.ResponseContentType())
-	}
+	// TODO: should only get the default pipeline since topic is empty, can we ensure this and use the passed context directly insatead of cloning, deal with error out here?
+	messageError := trigger.messageProcessor.MessageReceived(appContext, envelope, getResponseHandler(writer, lc))
 
-	_, err = writer.Write(appContext.ResponseData())
-	if err != nil {
-		lc.Errorf("unable to write ResponseData as HTTP response: %s", err.Error())
-		return
+	if messageError != nil {
+		// Process logs the error, so no need to log it here.
+		writer.WriteHeader(http.StatusInternalServerError)
+		_, _ = writer.Write([]byte(messageError.Error()))
 	}
+}
 
-	if appContext.ResponseData() != nil {
-		lc.Trace("Sent http response message", common.CorrelationHeader, correlationID)
+func getResponseHandler(writer http.ResponseWriter, lc logger.LoggingClient) interfaces.PipelineResponseHandler {
+	return func(ctx interfaces.AppFunctionContext, pipeline *interfaces.FunctionPipeline) error {
+		if len(ctx.ResponseContentType()) > 0 {
+			writer.Header().Set(common.ContentType, ctx.ResponseContentType())
+		}
+
+		_, err := writer.Write(ctx.ResponseData())
+		if err != nil {
+			lc.Errorf("unable to write ResponseData as HTTP response: %s", err.Error())
+			return err
+		}
+
+		if ctx.ResponseData() != nil {
+			lc.Trace("Sent http response message", common.CorrelationHeader, ctx.CorrelationID())
+		}
+		return nil
 	}
-
-	trigger.outputData = nil
```

Comment:
Where is this now being done?

Reply:
https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/internal/trigger/http/rest.go#L123

Looks like it was an artifact from before response data was moved into the context maybe?  Seems like only usage.

Reply:
Agreed. So the definition also needs to be removed form here:
https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/internal/trigger/http/rest.go#L46